crates/bevy_asset/src/server/mod.rs

```diff
@@ -214,19 +234,102 @@ impl AssetServer {
             .insert(handle_provider.type_id, handle_provider);
     }
 
+    /// Attempts to find the index of an [`AssetLoader`] matching the provided criteria
```

Comment:
If possible I think some tests would be nice to have

Reply:
Agreed.

Reply:
I've split `AssetLoaders` into it's own file to create some separation between it and the `AssetServer`. Doing this, I've moved all of the logic around loader registration/selection into `AssetLoaders`, and added several tests explicitly documenting how an `AssetLoader` is selected.